### PR TITLE
Standardize state diff stats in mdp datastore. Change variable name.

### LINKED ISSUE #112: Wrong scaling of 1-step diff_std, should be standardized
After having issues with my one variable having really high loss I realized that my issues stem from it having a very low value for diff_std in ar_model. This is the std-dev of 1-step differences. Digging a bit, I realize this is in mllam-data-prep computed for the non-standardized variables https://github.com/mllam/mllam-data-prep/blob/3a48c997144e66ef2478413668e669e124400cdd/mllam_data_prep/ops/statistics.py#L45 whereas it was earlier (and still for the MEPS datastore) for the standardized variables https://github.com/mllam/neural-lam/blob/f233f87c31ad7240644344f01e7cdb164044209e/neural_lam/datastore/npyfilesmeps/compute_standardization_stats.py#L283 These differences should be converted to the standardized scale before being used in the loss. It is not necessarily wrong to compute them for the non-standard vars, they just have to be re-scaled. And we should be consistent about the scaling for these that are returned from the datastore.

Having the wrong scaling of these is actually a significant problem, and lines like https://github.com/mllam/neural-lam/blob/f233f87c31ad7240644344f01e7cdb164044209e/neural_lam/models/base_graph_model.py#L174 will make it harder for the model rather than help. So this is something that needs fixing. 

I don't have any strong opinions about where the standardization should happen: 1. when computing the stats and storing them on disk 2. when loading these in the datastore or 3. in the model itself. We just need to decide on some strategy to keep this clear and consistent.

### PR DESCRIPTION
**Pull request by SimonKamuk**
## Describe your changes

This merge request updates the `MDPDatastore` to return `state_diff_std_standardized` and `state_diff_mean_standardized` in addition to the existing non-standardized versions. The `NpyFilesDatastoreMEPS` already only returned the standardized versions, so here the variable name has just been changed.

Also updated where `get_standardization_dataarray` is called to get the standardized versions.

## Issue Link

Closes #112 

## Type of change

- [x] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📖 Documentation (Addition or improvements to documentation)

## Checklist before requesting a review

- [ ] My branch is up-to-date with the target branch - if not update your fork with the changes from the target branch (use `pull` with `--rebase` option if possible).
- [ ] I have performed a self-review of my code
- [ ] For any new/modified functions/classes I have added docstrings that clearly describe its purpose, expected inputs and returned values
- [ ] I have placed in-line comments to clarify the intent of any hard-to-understand passages of my code
- [ ] I have updated the [README](README.MD) to cover introduced code changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have given the PR a name that clearly describes the change, written in imperative form ([context](https://www.gitkraken.com/learn/git/best-practices/git-commit-message#using-imperative-verb-form)).
- [ ] I have requested a reviewer and an assignee (assignee is responsible for merging). This applies only if you have write access to the repo, otherwise feel free to tag a maintainer to add a reviewer and assignee.

## Checklist for reviewers

Each PR comes with its own improvements and flaws. The reviewer should check the following:
- [ ] the code is readable
- [ ] the code is well tested
- [ ] the code is documented (including return types and parameters)
- [ ] the code is easy to maintain

## Author checklist after completed review

- [ ] I have added a line to the CHANGELOG describing this change, in a section
  reflecting type of change (add section where missing):
  - *added*: when you have added new functionality
  - *changed*: when default behaviour of the code has been changed
  - *fixes*: when your contribution fixes a bug

## Checklist for assignee

- [ ] PR is up to date with the base branch
- [ ] the tests pass
- [ ] author has added an entry to the changelog (and designated the change as *added*, *changed* or *fixed*)
- Once the PR is ready to be merged, squash commits and merge the PR.

### CODE REVIEW

**Comment by SimonKamuk** [comment]
I think this is ready for review. One question I have is whether I should keep both the normalized and non-normalized state_diff stats (this is what I've done), or if I should just remove them entirely.

**Comment by SimonKamuk** [comment]
Also, this goes in v0.4.0, right?

File: neural_lam/models/ar_model.py [hunk 1]
```diff
@@ -65,10 +65,12 @@ def __init__(
                 da_state_stats.state_std.values, dtype=torch.float32
             ),
             "diff_mean": torch.tensor(
```
Comment (joeloskarsson) [inline neural_lam/models/ar_model.py:67]: Just an additional comment, to be extra clear about this
```suggestion
            # Note that the one-step-diff stats (diff_mean and diff_std) are for differences 
            # computed on standardized data
            "diff_mean": torch.tensor(
```

File: neural_lam/datastore/mdp.py [hunk 2]
```diff
@@ -321,12 +322,27 @@ def get_standardization_dataarray(self, category: str) -> xr.Dataset:
...
         if category == "state":
             stats_variables.update(
                 {f"state__{split}__diff_{op}": f"state_diff_{op}" for op in ops}
             )
 
         ds_stats = self._ds[stats_variables.keys()].rename(stats_variables)
+
+        # Add standardized state diff stats
+        if category == "state":
+            ds_stats = ds_stats.assign(
```
Comment (joeloskarsson) [inline neural_lam/datastore/mdp.py:336]: I think it is enough to return only the standardized, I don't think we need the non-standardized version. And if I understand correctly the mdp datastore returns both non-standardized and standardized versions, while the npyfilesmeps returns only standardized. That is an inconsistency I would like to avoid.

**Review by joeloskarsson — CHANGES_REQUESTED** [review]
See small comments above (forgot to make them part of review)

File: neural_lam/datastore/mdp.py (hunk 2 above)
Comment (SimonKamuk) [reply neural_lam/datastore/mdp.py:330]: Done, removed the non-standardized diff stats.

**Comment by SimonKamuk** [comment]
Done 👍 care to have a final look and then I'll merge?

**Review by joeloskarsson — APPROVED** [review]
Perfect! lgtm
